### PR TITLE
magento/inventory#2348: [Store Pickup] Name in the FrontendName as null

### PR DESCRIPTION
**Pull request by max-tkachuk**
### Description (*)

1. Removed Pickup Location Name pre-processor
2. Added condition (IsPickupLocationActive and FrontendName is empty) to SaveInStorePickupPlugin.php when source Name should be used as location FrontName

### Fixed Issues (if relevant)

1. magento/inventory#2348: [Store Pickup] Name in the FrontendName as null

### Manual testing scenarios (*)

1. Go to BO > Stores > Inventory/Sources > Add New
2. Create source with empty FrontendName

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/inventory?pullRequest=2483) <br/>All committers have signed the CLA.

File: dev/tests/integration/etc/install-config-mysql.php.dist [hunk 1]
```diff
@@ -7,17 +7,13 @@
 return [
     'db-host' => 'localhost',
     'db-user' => 'root',
-    'db-password' => '123123q',
```
Comment (swnsma) [inline dev/tests/integration/etc/install-config-mysql.php.dist]: Please revert these changes.

**Review by swnsma — CHANGES_REQUESTED** [review]

File: dev/tests/integration/etc/install-config-mysql.php.dist (hunk 1 above)
Comment (max-tkachuk) [reply dev/tests/integration/etc/install-config-mysql.php.dist]: done

File: app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/SaveInStorePickupPlugin.php [hunk 2]
```diff
@@ -30,14 +44,26 @@ public function beforeSave(
...
 
-        if ($extensionAttributes !== null && $source instanceof  DataObject) {
-            $source->setData(Location::IS_PICKUP_LOCATION_ACTIVE, $extensionAttributes->getIsPickupLocationActive());
-            $source->setData(Location::FRONTEND_NAME, $extensionAttributes->getFrontendName());
-            $source->setData(Location::FRONTEND_DESCRIPTION, $extensionAttributes->getFrontendDescription());
+        if (empty($extensionAttributes->getFrontendName())) {
+            $extensionAttributes->setFrontendName($source->getName());
         }
 
+        $source->setData(Location::IS_PICKUP_LOCATION_ACTIVE, $extensionAttributes->getIsPickupLocationActive());
```
Comment (swnsma) [inline app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/SaveInStorePickupPlugin.php]: May we proceed with these steps only if Extension Attributes were passed?
Except the Frontend Name.

Extension Attributes Passed -> use Frontend Name from extension attributes.
Extension Attributes do not Passed -> Source does not have Frontend Name -> use Source Name
Extension Attributes do not Passed -> Source  has Frontend Name -> do not change the value

**Review by swnsma — CHANGES_REQUESTED** [review]

**Review by swnsma — APPROVED** [review]

**Review by novikor — APPROVED** [review]

**Comment by novikor** [comment]
@magento run Integration Tests build

**Comment by novikor** [comment]
@magento run Functional Tests build

**Review by novikor — CHANGES_REQUESTED** [review]
Fine, but the tests should be green

**Comment by swnsma** [comment]
@novikor,
Please review.
In case that everything is OK, PR can be merged.

**Review by ishakhsuvarov — APPROVED** [review]

**Review by novikor — APPROVED** [review]

**Pull request merged**